lib/reline/key_stroke.rb

```diff
@@ -1,8 +1,59 @@
 class Reline::KeyStroke
   using Module.new {
+    refine Integer do
+      def ==(other)
```

Comment:
Could we achieve this another way? Redefining `==` on integer (even in a refinement) will clear BOP_EQ for integers, making all comparisons slower.

ie. This optimization in MRI will no longer work https://github.com/ruby/ruby/blob/master/vm_insnhelper.c#L1986-L1987

Reply:
Terminal control software often spends more than 99% of its processing time displaying characters to the terminal. Do you have any performance measurement results that show that key processing is the bottleneck and should be improved?

Reply:
Sorry. The issue isn't in reline's processing, but that this will make Ruby's integer comparisons slower globally. All code will be slower, not just reline.

Here's a hacky demonstration benchmark on latest Ruby trunk:

```
jhawthorn@zergling:~/src/ruby (int_bop ) [ruby 3.1.0]
$ ruby -r 'benchmark' -e 'p Benchmark.realtime { 10_000_000.times { 1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1 } }'
1.1078672040021047

jhawthorn@zergling:~/src/ruby (int_bop ) [ruby 3.1.0]
$ ruby -r 'benchmark' -rreline -e 'p Benchmark.realtime { 10_000_000.times { 1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1;1==1 } }'
4.35402021300979
```

Reply:
> this will make Ruby's integer comparisons slower globally.

Thanks for the concise and clear explanation. I now understand the problem. It looks like `binding.irb` is going to cause serious problems in Rails applications.

Please wait a bit while we work on it.

Reply:
https://github.com/ruby/reline/pull/347
I did it. I think this problem was pretty bad. I'm glad I could handle it. Thanks again.

Reply:
Change looks great! Thank you for looking into this.